backend/src/main/java/com/bakdata/conquery/sql/execution/SqlExecutionService.java

```diff
@@ -76,20 +84,19 @@ private String getColumnName(ResultSet resultSet, int columnIndex) {
 		}
 	}
 
-	private Object[] getResultRow(ResultSet resultSet, int columnCount) {
-		// JDBC ResultSet indices start with 1 and we skip the first column because it contains the id
-		return IntStream.rangeClosed(2, columnCount)
-						.mapToObj(columnIndex -> this.getValueOfColumn(resultSet, columnIndex))
-						.toArray();
-	}
+	private SqlEntityResult getResultRow(ResultSet resultSet, List<ResultType> resultTypes, int columnCount) throws SQLException {
 
-	private String getValueOfColumn(ResultSet resultSet, int columnIndex) {
-		try {
-			return resultSet.getString(columnIndex);
-		}
-		catch (SQLException e) {
-			throw new ConqueryError.SqlError(e);
+		int rowNumber = resultSet.getRow();
+		String id = resultSet.getObject(PID_COLUMN_INDEX).toString();
+		Object[] resultRow = new Object[columnCount - 1];
+
+		for (int resultSetIndex = VALUES_OFFSET_INDEX; resultSetIndex <= columnCount; resultSetIndex++) {
+			int resultTypeIndex = resultSetIndex - VALUES_OFFSET_INDEX;
+			resultRow[resultTypeIndex] = ResultSetProcessor.getMapper(resultTypes.get(resultTypeIndex))
```

Comment:
die mapper bitte in einem parallelen array cachen

Reply:
Sollte hinfällig sein weil Teil der ResultType Klasse.

Reply:
> Sollte hinfällig sein weil Teil der ResultType Klasse.

👻

Reply:
@awildturtok Sorry, da muss ich nochmal nachfragen - die Mapper sind doch ein statisches Attribut vom `ResultSetProcessor` und werden nur einmal erzeugt? Oder meinst du ich soll die extra pro Execution cachen, sodass ich mir den getMapper()-Aufruf pro Result-Row spare?

Reply:
es reicht, dass in den caller von `getResultRow` vorzuberechnen. Der Code ist ziemlich hot, da wollen wir nicht Zeit verschwenden mit Map-lookups

Reply:
Alles klar 👍